### PR TITLE
Features/update unittests for verify reject or sign off

### PR DESCRIPTION
**Pull request by SamirHassoun**
## Updated Unit tests, Integration Tests

- Portal: VerifyTests
- WorkflowCoordinator:  PersistWorkflowInstanceDataCommandHandlerTests.cs, ProgressWorkflowInstanceEventHandlerTests.cs
- WorkflowCoordinator: Moved K2DbAssessmentTests.cs, CheckK2Connection  from Portal to WorkflowCoordinator for integration tests

## Code cleanup

- Portal: HttpClients Removed IWorkflowServiceApiClient.cs, WorkflowServiceApiClient.cs
- Portal: removed unused code: Startup.cs
- WorkflowCoordinator: removed unused code ProgressWorkflowInstanceEventHandler.cs

### CODE REVIEW

File: src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs [hunk 1]
```diff
@@ -22,6 +22,26 @@ public WorkflowServiceApiClient(HttpClient httpClient, IOptions<GeneralConfig> g
             _httpClient = httpClient;
         }
 
+        public async Task<bool> CheckK2Connection()
+        {
+            var fullUri = new Uri(_uriConfig.Value.K2WebServiceBaseUri, _uriConfig.Value.K2WebServiceGetWorkflowsUri);
+            string data;
```
Comment (benbhall) [inline src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:28]: This left from a previous iteration? You could probably just `var` it in the `using` now?

File: src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs [hunk 2]
```diff
@@ -22,6 +22,26 @@ public WorkflowServiceApiClient(HttpClient httpClient, IOptions<GeneralConfig> g
...
+
+            using (var response = await _httpClient.GetAsync(fullUri))
+            {
+                data = await response.Content.ReadAsStringAsync();
+
+                if (!response.IsSuccessStatusCode)
+                    throw new ApplicationException($"StatusCode='{response.StatusCode}'," +
```
Comment (benbhall) [inline src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:35]: I admit I don't know whole context but if this is a utility function, perhaps we return `bool` or a tuple to include any additional info and let caller decide whether to throw etc?

File: src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs (hunk 1 above)
Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs]: Fixed, thanks

File: src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs (hunk 2 above)
Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:34]: Thanks @benbhall , If the caller throw the error then we loose the call stack.

Comment (benbhall) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:34]: Just, if I call a checking function that returns a `bool`, I'd expect `true` or `false` not `true` or `exception`.

Comment (benbhall) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:34]: Caller knows why they called the function and what they want to do with it and stack will be intact to that call site.

Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs:34]: When throwing an exception its better done at the bottom method of the call chain to maintain the call stack if its related to that method's action.

I agree about the return being confusing, this particular method is only used in integration tests, so I'll change it to return false on failure

Comment (SamirHassoun) [reply src/WorkflowCoordinator/WorkflowCoordinator/HttpClients/WorkflowServiceApiClient.cs]: Fixed, thanks

**Pull request merged**